modules/pulsestorm/magento2/cli/library/module.php

```diff
@@ -17,24 +17,103 @@
 pestle_import('Pulsestorm\Cli\Code_Generation\createClassTemplate');
 pestle_import('Pulsestorm\Xml_Library\addSchemaToXmlString');
 pestle_import('Pulsestorm\Xml_Library\getXmlNamespaceFromPrefix');
+pestle_import('Pulsestorm\Pestle\Config\loadConfig');
+pestle_import('Pulsestorm\Pestle\Config\saveConfig');
 
-function getModuleInformation($module_name)
+function getAppCodePath() {
+    return 'app/code';
+}
+
+function getModuleAutoloaderPathFromComposerFile($module_name, $path_composer) {
+    if(!file_exists($path_composer)) {
+        throw new Exception("Could not find $path_composer");
+    }
+
+    $composer = json_decode(file_get_contents($path_composer));
+
+    $parts = explode('_', $module_name);
+    $moduleClassPrefix = $parts[0] . '\\' . $parts[1] . '\\';
+    $autoloadPath = false;
+    if(!isset($composer->autoload) || !isset($composer->autoload->psr4)) {
+        throw new Exception("No psr4 autoload section in $path_composer");
+    }
+    foreach($composer->autoload->psr4 as $prefix=>$path) {
+        if($prefix === $moduleClassPrefix) {
+            $autoloadPath = $path;
+        }
+    }
+    return rtrim($autoloadPath, '/');
+}
+
+/**
+ * if module is part of the package-folders config, then use the
+ * configured value.  Also, ensure that the folder we're pointing
+ * is actually part of the Magento system we're in
+ */
+function getModuleInformationFolderWhenConfigured($path_magento_base, $configured_path, $module_name) {
+    if(strpos($path_magento_base, $configured_path)) {
+        $message = "Configured Path is not in Magento folder\n" .
+            "Path: ".$configured_path."\n" .
+            "Magento Path: ".$path_magento_base."\n\n";
+        throw new Exception($message);
+    }
+
+    // find composer autoload path
+    $pathComposer = $configured_path . '/composer.json';
+    $autoloadPath = getModuleAutoloaderPathFromComposerFile($module_name, $pathComposer);
+
+    if(!$autoloadPath) {
+        throw new Exception("Could not find autoload path in $pathComposer");
+    }
+
+    return rtrim(preg_replace(
+        '%' . $path_magento_base . '/%', '', $configured_path, 1
+    ), '/');
+}
+
+function getModuleInformation($module_name, $path_magento_base=false)
 {
-    list($vendor, $name) = explode('_', $module_name);        
-    return (object) [
+    $path_magento_base = $path_magento_base ? $path_magento_base : getBaseMagentoDir();
+
+    list($vendor, $name) = explode('_', $module_name);
+    $information = [
         'vendor'        => $vendor,
         'short_name'    => $name,
         'name'          => $module_name,
-        'folder'        => getBaseMagentoDir() . "/app/code/$vendor/$name",
     ];
+
+
+    // we need to check the configuration for a path.  If it exists, then
```

Comment:
Lots of shenanagins to make sure we're loading the right path depending on whether the user's working out of app/code or a vendor folder.